### PR TITLE
feat: show generic error screen

### PR DESCRIPTION
**Pull request by mcgarj**
# DCMAW-7181: User sees error screen in app when errors occur (generic, catch-all)

This PR shows a generic error screen when an error is thrown during the login authentication process. 

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update
- [x] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [x] Met all accessibility requirements?
    - [x] Checked dynamic type sizes are applied
    - [x] Checked VoiceOver can navigate your new code
    - [x] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [ ] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

**Review by DylanRJ — APPROVED** [review]

File: OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -41,8 +32,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/firebase/firebase-ios-sdk.git",
       "state" : {
-        "revision" : "5de0369ee79ad096c164eb3afeb7921d92a43b58",
-        "version" : "10.18.0"
+        "revision" : "8872dbd7d947acf757abab933da10e83c1842280",
```
Comment (benmckillop) [inline OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved:35]: Looks like these have all been downgraded, worth upgrading locally and pushing again :)

File: Sources/Errors/GenericErrorViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import GDSAnalytics
+import GDSCommon
+import Logging
+
+struct GenericErrorViewModel: GDSErrorViewModel, BaseViewModel {
+    let image: String = "exclamationmark.circle"
+    let title: GDSLocalisedString = "Something went wrong"
```
Comment (benmckillop) [inline Sources/Errors/GenericErrorViewModel.swift:7]: Should these strings be localised?

File: OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (DylanRJ) [reply OneLogin.xcworkspace/xcshareddata/swiftpm/Package.resolved:35]: Oh good spot, dno why that happened, will change locally 👍

File: Sources/Errors/GenericErrorViewModel.swift (hunk 2 above)
Comment (DylanRJ) [reply Sources/Errors/GenericErrorViewModel.swift:7]: Ideally yes, but haven't got our ticket to implement localisation yet 🙁

**Review by benmckillop — APPROVED** [review]

**Pull request merged**